lib/utils/authorizeSilently.ts

```diff
@@ -0,0 +1,52 @@
+import { SilentAuthorizationError } from "../errors";
+import { TimeoutID } from "../types";
+
+const SERVER_MESSAGE_TIMEOUT = 10000;
+
+export const authorizeSilently = (uri: string, origin: string) => {
+  return new Promise((resolve, reject) => {
+    let timeoutId: TimeoutID;
+    const iframe = document.createElement("iframe");
+    iframe.style.display = "none";
+    document.body.appendChild(iframe);
+
+    const cleanup = () => {
+      window.removeEventListener("message", onMessageHandler);
+      if (iframe && iframe.parentNode) {
+        iframe.parentNode.removeChild(iframe);
+      }
+      if (timeoutId) {
+        clearTimeout(timeoutId);
+      }
+    };
+
+    const onMessageHandler = (event: MessageEvent) => {
+      if (event.origin !== origin) {
+        return;
+      }
+
+      cleanup();
+
+      if (!event.data.error) {
+        const { code } = event.data.response;
+        return resolve(code);
+      } else {
+        return reject(event.data.error);
+      }
+    };
+
+    window.addEventListener("message", onMessageHandler);
+    if (iframe) {
+      iframe.src = uri;
+    }
+
+    timeoutId = setTimeout(() => {
```

Comment:
does it seem useful to timeout after some time if server doesn't send message? is 10 secs appropriate?

Reply:
Yes, it's totally make sense. 
Should we make it configurable to be sure that it will be appropriate for app that uses it?

Reply:
Yeah I agree, good idea 🙂 I have added a parameter for this to the method: 3ca8ece9625143e2009d047a254beeaa4b81c4fc